backfill/2020-01-23-sync-ping/README.md

```diff
@@ -0,0 +1,115 @@
+# Backfilling sync pings due to unexpected devices fields
+
+See https://bugzilla.mozilla.org/show_bug.cgi?id=1610907
+
+## Steps
+
+We ran the backfill in the `moz-fx-data-backfill-4` project. 
+
+First, we determine the backfill range by querying the relevant error table:
+
+```sql
+SELECT
+  date(submission_timestamp) as dt, count(*)
+FROM
+  `moz-fx-data-shared-prod.payload_bytes_error.telemetry`
+WHERE
+  DATE(submission_timestamp) >= "2020-01-10"
+  AND document_type = 'sync'
+  AND error_message LIKE 'org.everit.json.schema.ValidationException: #/payload/devices/%: extraneous key % is not permitted'
+  group by 1 order by 1
+```
+
+That showed the affected range as `2020-01-20` through `2020-01-23`.
+
+Next, we create destination tables via the `mirror-prod-tables` script and
+we populate the `backfill_input` table via query:
+
+```sql
+SELECT
+  *
+FROM
+  `moz-fx-data-shared-prod.payload_bytes_error.telemetry`
+WHERE
+  DATE(submission_timestamp) BETWEEN "2020-01-20" AND "2020-01-23"
+  AND document_type = 'sync'
+  AND error_message LIKE 'org.everit.json.schema.ValidationException: #/payload/devices/%: extraneous key % is not permitted'
+```
+
+Next, we construct a suitable Dataflow job configuration in
+`launch-dataflow-sync-ping` and run the script.
+
+It may be useful to do a quick run with a subset of data in order to make
+sure the scripts are all properly configured by writing a subset of the backfill
+input to a table and then running the script on the subset:
+
+```sql
+SELECT
+  *
+FROM
+  `moz-fx-data-backfill-4.payload_bytes_error.backfill_input`
+LIMIT
+  1000
+```
```

Comment:
That's a good idea. Did you have to manually empty tables after having a successful test?

Reply:
Yes I had to manually delete.  I'll note that here